### PR TITLE
sys-process/iotop: add 0.6_p20230124

### PR DESCRIPTION
**Pull request by joecool1029**
This bump fixes a lot of lingering issues as well as removing the need for patching. Moved to EAPI8 and PEP517.

Closes: https://github.com/gentoo/gentoo/pull/24258
Closes: https://bugs.gentoo.org/815151
Signed-off-by: Joe Kappus <joe@wt.gd>

### CODE REVIEW

File: sys-process/iotop/iotop-0.6_p20230124.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+RDEPEND="!sys-process/iotop-c"
+
+CONFIG_CHECK="~TASK_IO_ACCOUNTING ~TASK_DELAY_ACCT ~TASKSTATS ~VM_EVENT_COUNTERS"
+
+DOCS=( NEWS README THANKS )
+
+S="${WORKDIR}/${PN}-a14256a"
+
+pkg_setup() {
```
Comment (thesamesam) [inline sys-process/iotop/iotop-0.6_p20230124.ebuild:29]: Drop, it'll be picked up by default as no other inherited eclass exports pkg_setup, so it'll be the same if you don't define it.

File: sys-process/iotop/iotop-0.6_p20230124.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	linux-info_pkg_setup
+}
+
+pkg_postinst() {
+	ewarn "Since Linux 5.14, sysctl kernel.task_delayacct should be enabled"
+	ewarn "This can be enabled by running: 'sysctl kernel.task_delayacct=1' "
+	ewarn "And can be made persistent by adding 'kernel.task_delayacct = 1' to /etc/sysctl.conf"
```
Comment (thesamesam) [inline sys-process/iotop/iotop-0.6_p20230124.ebuild:36]: Maybe only show this once with the readme eclasses? Or leave it for now and we can revisit, I guess.
```suggestion
	ewarn "And can be made persistent by adding 'kernel.task_delayacct = 1' to ${EROOT}/etc/sysctl.conf"
```

File: sys-process/iotop/iotop-0.6_p20230124.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
+
+RDEPEND="!sys-process/iotop-c"
+
+CONFIG_CHECK="~TASK_IO_ACCOUNTING ~TASK_DELAY_ACCT ~TASKSTATS ~VM_EVENT_COUNTERS"
+
+DOCS=( NEWS README THANKS )
+
+S="${WORKDIR}/${PN}-a14256a"
```
Comment (thesamesam) [inline sys-process/iotop/iotop-0.6_p20230124.ebuild:27]: Let's put this below SRC_URI (directly below) and also change it so it truncates the first N characters from MY_COMMIT, so you don't have to manually sync it in future.

File: sys-process/iotop/iotop-0.6_p20230124.ebuild [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_REQ_USE="ncurses(+)"
+
+MY_COMMIT="a14256a3ff74eeee59493ac088561f1bafab85a7"
+inherit distutils-r1 linux-info
+
+DESCRIPTION="Top-like UI used to show which process is using the I/O"
+HOMEPAGE="http://guichaz.free.fr/iotop/"
+SRC_URI="https://repo.or.cz/iotop.git/snapshot/${MY_COMMIT}.tar.gz -> ${P}.tgz"
```
Comment (thesamesam) [inline sys-process/iotop/iotop-0.6_p20230124.ebuild:15]: Let's keep suffix the same:
```suggestion
SRC_URI="https://repo.or.cz/iotop.git/snapshot/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
```

File: sys-process/iotop/iotop-0.6_p20230124.ebuild (hunk 3 above)
Comment (joecool1029) [reply sys-process/iotop/iotop-0.6_p20230124.ebuild:27]: I knew I should have done this, but I don't know how to do it. Couldn't find any examples in ::gentoo either.

Comment (joecool1029) [reply sys-process/iotop/iotop-0.6_p20230124.ebuild]: hairu or whatever he calls himself now showed me how to do this.

**Pull request closed without merging**